modules/metastorage/README.md

```diff
@@ -0,0 +1,46 @@
+# Metastorage
+
+The module for store and access to metadata. There is module is connected with this one:
+
+- metastorage-api - the module contains classes to access to metastorage service from another components.
+
+The storage is distributed to prevent losing data and based of RAFT. That distributed storage persists on top of every cluster node, however
+most of such participants are learners and only few of them are voting ones. In typical case, the group consists of only several voted nodes
+of the cluster (amount of nodes have to be odd 3 or 5). The rest of nodes listens metadata update but does not participant in voting (RAFT 
+learners).
+
+## Threading model
+
+There is only one dedicate thread to notify watches. This is a thread with prefix metastorage-watch-executor.
+The thread is created from executor in KeyValueStorage:
+
+```java
+private ExecutorService watchExecutor = Executors.newSingleThreadExecutor(new NamedThreadFactory(NamedThreadFactory
+        .threadPrefix(nodeName, "metastorage-watch-executor"),LOG));
+```
+
+Also, the storage internally contains a two threads executor to create a snapshot:
+
+```java
+private ExecutorService snapshotExecutor = Executors.newFixedThreadPool(2,new NamedThreadFactory(NamedThreadFactory
+        .threadPrefix(nodeName, "metastorage-snapshot-executor"),LOG));
+```
+
+Both of the executors have a node name in their prefixes to distinguish to which node the particular thread belongs.
+
```

Comment:
To distinguish to which node the specific thread belongs, each executor has a node name in its prefix.